cdap-proto/src/main/java/co/cask/cdap/proto/ProtoConstraint.java

```diff
@@ -112,7 +118,16 @@ public long getMillisAfterTrigger() {
     }
 
     @Override
+    public void setWaitUntilMet(boolean waitUntilMet) {
+      super.setWaitUntilMet(waitUntilMet);
```

Comment:
I wanted to keep the logic of validation within the `validate` method only. So I need to call the super's setter method so that `validate` sees the new value of the object.